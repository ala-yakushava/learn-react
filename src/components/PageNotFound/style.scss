@import "../../styles/theme";

.PageNotFound {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  padding-top: 20px;

  &_inner {
    padding: 150px 0 200px;
  }

  &_title {
    text-align: center;
  }

  &_404 {
    margin: 50px 0 80px;
    font-size: 200px;
    line-height: 1;
    font-weight: 700;
    color: $red;
    letter-spacing: 20px;
    text-align: center;
  }

  &_button {
    display: block;
    margin: 0 auto;
  }
}
