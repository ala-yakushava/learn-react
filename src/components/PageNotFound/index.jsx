import React from 'react';

import './style.scss';
import Container from '../Container';
import Logo from '../Logo';
import Heading from '../Heading';
import Button from '../Button';
import Footer from '../Footer';

const PageNotFound = () => (
  <div className="PageNotFound">
    <Container>
      <Logo />
      <div className="PageNotFound_inner">
        <Heading className="PageNotFound_title">Page Not Found</Heading>
        <p className="PageNotFound_404">404</p>
        <Button mode="secondary" className="PageNotFound_button">Go back to home</Button>
      </div>
    </Container>
    <Footer />
  </div>
);

export default PageNotFound;
