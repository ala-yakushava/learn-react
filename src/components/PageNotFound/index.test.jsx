import React from 'react';
import renderer from 'react-test-renderer';

import PageNotFound from './index';

describe('Components - PageNotFound', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(<PageNotFound />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
