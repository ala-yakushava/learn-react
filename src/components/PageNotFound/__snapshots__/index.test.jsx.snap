// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - PageNotFound should match snapshot 1`] = `
<div
  className="PageNotFound"
  data-testid="page-not-found"
>
  <div
    className="Container"
  >
    <p
      className="Logo"
    >
      <b>
        netflix
      </b>
      roulette
    </p>
    <div
      className="PageNotFound_inner"
    >
      <div
        className="Heading PageNotFound_title"
      >
        Page Not Found
      </div>
      <p
        className="PageNotFound_404"
      >
        404
      </p>
      <button
        className="Button Button--secondary PageNotFound_button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="Button_text"
        >
          Go back to home
        </span>
      </button>
    </div>
  </div>
  <footer
    className="Footer"
  >
    <p
      className="Logo"
    >
      <b>
        netflix
      </b>
      roulette
    </p>
  </footer>
</div>
`;
